src/axom/quest/MeshViewUtil.hpp

```diff
@@ -0,0 +1,778 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_MESH_VIEW_UTIL_H_
+#define QUEST_MESH_VIEW_UTIL_H_
+
+#include "axom/config.hpp"
+
+// Implementation requires Conduit.
+#ifdef AXOM_USE_CONDUIT
+
+  #include "axom/core.hpp"
+  #include "axom/fmt.hpp"
+  #include "conduit_blueprint.hpp"
+  #include "conduit_blueprint_mcarray.hpp"
+  #ifdef AXOM_USE_MPI
+    #include "conduit_blueprint_mpi.hpp"
+    #include "conduit_relay_mpi_io_blueprint.hpp"
+  #endif
+
+  #include <memory>
+  #include <limits>
+  #include <cstdlib>
+  #include <cmath>
+  #include <vector>
+
+namespace axom
+{
+namespace quest
+{
+namespace internal
+{
+template <typename T>
+constexpr T BADINDEX = std::numeric_limits<T>::max();
+template <typename T, int DIM>
+inline axom::StackArray<T, DIM> makeStackArray(T v = std::numeric_limits<T>::max())
+{
+  axom::StackArray<T, DIM> rval;
+  for(int d = 0; d < DIM; ++d)
+  {
+    rval[d] = v;
+  }
+  return rval;
+}
+template <typename T, int DIM, typename U>
+inline axom::StackArray<T, DIM> makeStackArray(const U* v)
+{
+  axom::StackArray<T, DIM> rval;
+  for(int d = 0; d < DIM; ++d)
+  {
+    rval[d] = v[d];
+  }
+  return rval;
+}
+}  // namespace internal
+
+/**
+   \brief Utility for high-level access into a blueprint mesh,
+   for structured mesh with explicit coordinates.
+
+   Blueprint mesh data is sufficient but sparse, leaving users
+   to compute a number of intermediate data to get to high-level
+   data of interest, such as views into array data.  This class
+   encapsulates those common utilities.
+
+   Views are single-domain-specific.  They don't apply to multi-domain
+   meshes.  They are also topology specific, with the topology name
+   given to the constructor.
+
+   TODO: Figure out if there's a better place for this utility.
+   It's only in axom/quest because the initial need was there.
+*/
+template <int DIM, axom::MemorySpace MemSpace>
+class MeshViewUtil
+{
+public:
+  using CoordsViewsType =
+    axom::StackArray<axom::ArrayView<double, DIM, MemSpace>, DIM>;
+  using ConstCoordsViewsType =
+    axom::StackArray<axom::ArrayView<const double, DIM, MemSpace>, DIM>;
+
+  //@{
+  //@name Setting up
+  /*!
+    @brief Constructor
+    @param [in] bpDomain Blueprint single domain.
+    @param [in] topologyName Name of topology in the domain.
+
+    If \a topologyName is omitted, use the first topology,
+
+    The topology dimension must match DIM.
+  */
+  MeshViewUtil(conduit::Node& bpDomain, const std::string& topologyName = "")
+    : m_dom(&bpDomain)
+    , m_topologyName(topologyName)
+  {
+    if(m_topologyName.empty())
+    {
+      m_topologyName = bpDomain.fetch_existing("topologies")[0].name();
```

Comment:
Should we handle the case where there are topologies?

Reply:
Perhaps that handled in the `isValid()` call below ?

Reply:
Not sure I understand.  If you're asking whether we handle multiple topologies in the same domain, we do--by the user specifying one of those topologies.

Reply:
_(Sorry -- there was a missing word that changed the meaning! I corrected it above)_

I was asking if we should explicitly handle the case where there are \*no\* topologies in the provided blueprint, or if this is sufficiently handled in the `isValid()` call at the bottom of the function.

Reply:
It would be weird if there are no topologies.  I think it will fail the blueprint validity test and consequently `isValid()`.